### PR TITLE
Delete some unused methods & fix typos

### PR DESCRIPTION
**Pull request by radito3**

### CODE REVIEW

File: commands/base_command.go [hunk 1]
```diff
@@ -128,13 +128,8 @@ func GetOptionValue(args []string, optionName string) string {
...
 // NewRestClient creates a new MTA deployer REST client
-func (c *BaseCommand) NewRestClient(host string) (restclient.RestClientOperations, error) {
-	restClient := c.clientFactory.NewRestClient(host, c.transport, c.jar, c.tokenFactory)
-	return restClient, nil
-}
-
-func (c *BaseCommand) NewManagementRestClient(host string) (restclient.RestClientOperations, error) {
-	return c.clientFactory.NewManagementRestClient(host, c.transport, c.jar, c.tokenFactory), nil
+func (c *BaseCommand) NewRestClient(host string) restclient.RestClientOperations {
+	return c.clientFactory.NewRestClient(host, c.transport, c.jar, c.tokenFactory)
```
Comment (theghost5800) [inline commands/base_command.go:132]: As I can see the practise is to return client and nil as second value which indicate that there is no errors. Take a look at NewMtaClient and NewMtaV2Client functions

File: commands/purge_config_command.go [hunk 2]
```diff
@@ -52,11 +52,7 @@ func (c *PurgeConfigCommand) Execute(args []string) ExecutionStatus {
 		terminal.EntityNameColor(context.Space),
 		terminal.EntityNameColor(context.Username))
 
-	rc, err := c.NewRestClient(host)
-	if err != nil {
-		c.reportError(baseclient.NewClientError(err))
-		return Failure
-	}
+	rc := c.NewRestClient(host)
```
Comment (theghost5800) [inline commands/purge_config_command.go:55]: Are you sure that it is not possible to return error while creating client?

File: util/archive_builder.go [hunk 3]
```diff
@@ -164,18 +164,18 @@ func copyDirectory(src, dest string) error {
 
 		if fd.IsDir() {
 			if err = copyDirectory(srcfp, dstfp); err != nil {
-
+				return err
```
Comment (theghost5800) [inline util/archive_builder.go:167]: It will be good to have unit test about this.

File: commands/base_command.go (hunk 1 above)
Comment (radito3) [reply commands/base_command.go:132]: I don't think it's a "convention" per se, it's just that, in the `NewMtaClient` and `NewMtaV2Client` methods, there is an operation that returns an error (the `GetSpace` call), whereas here, an error can't be returned when creating the client.

File: commands/purge_config_command.go (hunk 2 above)
Comment (radito3) [reply commands/purge_config_command.go:55]: Yes.

File: util/archive_builder.go (hunk 3 above)
Comment (radito3) [reply util/archive_builder.go:167]: There _are_ unit tests for `archive_handler` that handle errors when working with files

Comment (theghost5800) [reply util/archive_builder.go:167]: I mean that after you start to return error, some test should be start to fail. If it's not probably there is no test for such cases.

**Review by theghost5800 — APPROVED** [review]

**Pull request merged**